scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2631,6 +2632,14 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
               targs.fold(ref)(x => autoEndPos(sid)(Term.ApplyType(ref, x))),
               autoPos(argumentPatterns().reduceWith(Pat.ArgClause.apply))
             )
+          } else if (at[Equals] && isNamedTupleOk) {
```

Comment:
Should we check dialect here? All other cases are already covered and parse.

Reply:
if there's no confusion (`=` can't be interpreted differently), i am in favor of skipping as otherwise you'd have to error out if the dialect doesn't allow.

unrelated: see below for an optional implementation alternative.